### PR TITLE
Tp data tool

### PR DESCRIPTION
**Pull request by phlax**
adds tp data tool for setting tp stats data

### CODE REVIEW

File: tests/pootle_data/data_tool.py [hunk 1]
```diff
@@ -160,15 +160,8 @@ def test_data_tool_tp_get_stats_with_children(tp0):
 
 
 @pytest.mark.django_db
-def test_data_tool_store_get_checks(store0):
```
Comment (phlax) [inline tests/pootle_data/data_tool.py]: i need to check the commits, its been rebased a lot

File: pootle/apps/pootle_data/utils.py [hunk 2]
```diff
@@ -231,3 +239,163 @@ def save_data(self):
...
+
+    def get_child_stats(self, children, child, root=None, use_aggregates=True):
+        root = root or self.get_root(child)
+        mapping = dict(
+            total_words="total",
+            fuzzy_words="fuzzy",
+            translated_words="translated",
+            critical_checks="critical",
+            pending_suggestions="suggestions")
+        if root not in children:
```
Comment (phlax) [inline pootle/apps/pootle_data/utils.py]: this could be a bit simpler

File: pootle/apps/pootle_data/utils.py [hunk 3]
```diff
@@ -231,3 +239,163 @@ def save_data(self):
...
+                fuzzy=0,
+                translated=0,
+                suggestions=0)
+        for k in self.sum_fields:
+            child_k = k
+            if use_aggregates:
+                child_k = "%s__sum" % k
+            children[root][mapping.get(k, k)] += child[child_k]
+        for k in self.max_fields:
+            child_k = k
```
Comment (phlax) [inline pootle/apps/pootle_data/utils.py]: simplify

File: pootle/apps/pootle_data/utils.py [hunk 4]
```diff
@@ -231,3 +239,163 @@ def save_data(self):
...
+            if use_aggregates:
+                child_k = "%s__max" % k
```
Comment (unho) [inline pootle/apps/pootle_data/utils.py]: child_k += "__max"

Comment (phlax) [inline pootle/apps/pootle_data/utils.py]: interpolation is generally preferable

**Pull request merged**